### PR TITLE
Prepare for Rails 7.1's error reporter API change

### PR DESCRIPTION
**Pull request by st0012**
Rails 7.1 will introduce a new `source` keyword argument to `ErrorReporter` APIs. So we should support that as well.

Related PR: https://github.com/rails/rails/pull/44998

### CODE REVIEW

**Comment by st0012** [comment]
cc @casperisfine

File: sentry-rails/lib/sentry/rails/error_subscriber.rb [hunk 1]
```diff
@@ -3,8 +3,17 @@ module Rails
...
     class ErrorSubscriber
-      def report(error, handled:, severity:, context:)
-        Sentry::Rails.capture_exception(error, level: severity, contexts: { "rails.error" => context }, tags: { handled: handled })
+      SKIP_SOURCES = [/.*_cache_store.active_support/]
+
+      def report(error, handled:, severity:, context:, source: nil)
+        tags = { handled: handled }
+
+        if source
+          return if SKIP_SOURCES.any? { |skip_source| skip_source.match?(source) }
```
Comment (casperisfine) [inline sentry-rails/lib/sentry/rails/error_subscriber.rb:12]:
```suggestion
      SKIP_SOURCES = Regexp.union([/.*_cache_store.active_support/])

      def report(error, handled:, severity:, context:, source: nil)
        tags = { handled: handled }

        if source
          return if SKIP_SOURCES.match?(source)
```

**Review by casperisfine — COMMENTED** [review]
❤️

File: sentry-rails/lib/sentry/rails/error_subscriber.rb (hunk 1 above)
Comment (st0012) [reply sentry-rails/lib/sentry/rails/error_subscriber.rb:12]: Never knew this usage before, thanks!

**Review by sl0thentr0py — APPROVED** [review]

**Pull request merged**